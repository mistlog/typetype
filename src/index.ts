export * from "./parser";
export * from "./generator";
export * from "./core";