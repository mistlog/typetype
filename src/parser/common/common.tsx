import { ReactPeg } from "react-peg";

// white space
export function _() {
    return (
        <repeat type="*">
            <set> \t\n\r</set>
        </repeat>
    );
}

export const Text = (content: string) => {
    return (
        <list>
            <_ /><text>{content}</text><_ />
        </list>
    )
}