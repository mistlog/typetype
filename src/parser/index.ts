export * from "./common";
export * from "./basic";
export * from "./function";
export * from "./expression";
export * from "./statement";
export * from "./declaration";
export * from "./start";