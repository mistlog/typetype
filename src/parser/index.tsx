export * from "./common";
export * from "./basic";
export * from "./expression";
export * from "./statement";
export * from "./declaration";
export * from "./start";