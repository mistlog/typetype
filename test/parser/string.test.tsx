import { ReactPeg } from "react-peg";
import { loadType, saveAST } from "../common";
import { StringTypeLiteral } from "../../src";

describe("parser: string", ()=>{
    test("String: special char", () => {
        const parser = ReactPeg.render(<StringTypeLiteral />);
        const name = "String-SpecialChar";
        const code = loadType(name);

        const ast = parser.parse(code);
        saveAST(ast, `${name}.json`);
        expect(ast).toMatchSnapshot();
    })
})