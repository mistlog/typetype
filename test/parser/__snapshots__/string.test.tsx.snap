// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser: string String: special char 1`] = `
Object {
  "kind": "StringTypeLiteral",
  "value": "\\\\n",
}
`;
